Akka.Serialization.MessagePack/Internal/IntIndexedMessagePackFormatterDict.cs

```diff
@@ -50,32 +50,39 @@ public IMessagePackFormatter TryAdd(int i, IMessagePackFormatter formatter)
         //However this will not impact types that can already be deserialized,
         //and since we are always in a full fence here can ensure we don't
         //accidentally fail to share an instance.
-        
-        lock (_lockObj)
+        if (i >= 0)
```

Comment:
To be clear, We _should_ always be in this condition, unless somebody did something wrong somewhere else.
Still, better to guard than risk infinite-looping till we OOM growing :)